### PR TITLE
API-1835: Make a useful example and add operator-integration testing for it

### PR DESCRIPTION
**Pull request by deads2k**
this adds `make test-operator-integration` that we'll use to verify our sample operator works well.  it's a very basic example that we can make do operator things.

This also adds to the MOM things that are useful when migrating an existing operator to MOM and the existing operator.

Several bugs were found.  Also requires https://github.com/openshift/library-go/pull/1834.

Some basic control loops will be nice to have.  probably versioning too.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,3 +41,8 @@ test-e2e:
 	hack/e2e.sh
 
 .PHONY: test-e2e
+
+test-operator-integration: build
```
Comment (p0lyn0mial) [inline Makefile:45]: would be useful to run in a CI job.

File: pkg/sampleoperator/sampleapplyconfiguration/sample_apply_configuration.go [hunk 2]
```diff
@@ -2,11 +2,190 @@ package sampleapplyconfiguration
...
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
 func SampleRunApplyConfiguration(ctx context.Context, input libraryapplyconfiguration.ApplyConfigurationInput) (libraryapplyconfiguration.AllDesiredMutationsGetter, error) {
```
Comment (p0lyn0mial) [inline pkg/sampleoperator/sampleapplyconfiguration/sample_apply_configuration.go:29]: How do we know if it works ?

File: pkg/applyconfiguration/execer.go [hunk 3]
```diff
@@ -32,7 +33,11 @@ func ExecApplyConfiguration(ctx context.Context, binaryPath, inputDirectory, out
 		"--output-dir", outputDirectory,
 	}
 
-	cmd := exec.CommandContext(ctx, binaryPath, args...)
+	// TODO prove that the timeout works if the process captures sig-int
```
Comment (p0lyn0mial) [inline pkg/applyconfiguration/execer.go:36]: maybe the applyconfig library could offer sigterm handling

File: pkg/sampleoperator/sampleapplyconfiguration/sample_apply_configuration.go (hunk 2 above)
Comment (deads2k) [reply pkg/sampleoperator/sampleapplyconfiguration/sample_apply_configuration.go:29]: this function?  returns an error and will have `make test-operator-integration`

File: pkg/applyconfiguration/execer.go (hunk 3 above)
Comment (deads2k) [reply pkg/applyconfiguration/execer.go:36]: > maybe the applyconfig library could offer sigterm handling

Probably reasonable, but not here.

**Pull request merged**